modules/oe_content_featured_media_field/tests/src/FunctionalJavascript/FeaturedMediaFieldWidgetTest.php

```diff
@@ -0,0 +1,226 @@
+      ->getViewDisplay($this->field->getTargetEntityTypeId(), $this->field->getTargetBundle())
+      ->setComponent($this->fieldStorage->getName(), $view_display_options);
+    $display->save();
+  }
+
+  /**
+   * Tests the featured media widget.
+   */
+  public function testFeaturedMediaWidget(): void {
+    // Visit the node add page.
+    $this->drupalGet('node/add/page');
+    $page = $this->getSession()->getPage();
+    $assert_session = $this->assertSession();
+
+    // Assert that the fields and labels are present on the page.
+    $assert_session->pageTextContains('Featured media');
+    $assert_session->fieldExists('Media item');
+    $assert_session->fieldExists('Caption');
+
+    // Assert the help texts without media overview permission.
+    $assert_session->pageTextContains('Start typing the name of the Media.');
+    $assert_session->pageTextNotContains('You can manage all the media items on this page.');
+    $assert_session->pageTextContains('Allowed media types: Image, Document');
+
+    // Login with a user with extended permissions.
+    $this->drupalLogin($this->drupalCreateUser([
+      'access content',
+      'access media overview',
+      'create image media',
+      'create page content',
+    ]));
+    $this->drupalGet('node/add/page');
+
+    // Assert the help texts with media overview permission.
+    $assert_session->pageTextContains('Start typing the name of the Media.');
+    $assert_session->pageTextContains('You can manage all the media items on this page.');
+    $assert_session->pageTextContains('Allowed media types: Image, Document');
+
+    // Test that the Media item field turns required once the Caption is filled.
+    $this->assertFalse($page->findField('Media item')->hasAttribute('required'));
+    $page->fillField('Caption', 'Caption text');
+    $this->assertTrue($page->findField('Media item')->hasAttribute('required'));
+
+    // Test the autocomplete functionality returns the created media items.
+    $this->doAutocomplete($this->fieldStorage->getName(), 'Test');
+    $results = $page->findAll('css', '.ui-autocomplete li');
+    $this->assertCount(2, $results);
+    $assert_session->pageTextContains('Test image');
+    $assert_session->pageTextContains('Test media');
+
+    // Assign the image media and save the node.
+    $page->fillField('Media item', 'Test image');
+    $page->fillField('Title', 'My test node');
+    $page->pressButton('Save');
+
+    // Assert the label and values are visible on the node page.
+    $assert_session->pageTextContains('Featured media');
+    $assert_session->pageTextContains('Test image');
+    $assert_session->pageTextContains('Caption text');
+  }
+
+  /**
+   * Executes an autocomplete on a given field and waits for it to finish.
+   *
+   * @param string $field_name
+   *   The field name.
+   * @param string $value
+   *   The value to look for.
+   */
+  protected function doAutocomplete(string $field_name, string $value): void {
```

Comment:
The field thought needs to be a featured media one, since we add `[0][featured_media][target_id]`. So the name is a bit misleading. We use it only once, we can merge it in the main method too. Or rename this better.